functions/Get-DbaDatabase.ps1

```diff
@@ -0,0 +1,174 @@
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Get-DbaDatabase
+
+.EXAMPLE
+Get-DbaDatabase -SqlServer localhost
+Returns all databases on the local default SQL Server instance
+
+.EXAMPLE
+Get-DbaDatabase -SqlServer localhost -System
+Returns only the system databases on the local default SQL Server instance
+
+.EXAMPLE
+Get-DbaDatabase -SqlServer localhost -User
+Returns only the user databases on the local default SQL Server instance
+	
+.EXAMPLE
+'localhost','sql2016' | Get-DbaDatabase
+Returns databases on multiple instances piped into the function
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Position = 0, Mandatory = $true, ValueFromPipeline = $True)]
+		[Alias("ServerInstance", "SqlServer")]
+		[object[]]$SqlInstance,
+		[System.Management.Automation.PSCredential]$SqlCredential,
+		[switch]$System,
+		[switch]$User,
+        [switch]$Owner,
+        [switch]$Encrypted,
+		[ValidateSet('Online', 'Offline', 'Emergency')]
+		[string]$State,
+		[ValidateSet('ReadOnly', 'ReadWrite')]
+		[string]$Access,
+        [ValidateSet('Full', 'Simple', 'BulkLogged')]
+        [string]$RecoveryModel
+	)
+	
+	DynamicParam { if ($SqlInstance) { return Get-ParamSqlDatabases -SqlServer $SqlInstance[0] -SqlCredential $SqlCredential } }
+	
+	BEGIN
+	{
+		$databases = $psboundparameters.Databases
+	}
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlInstance)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $instance -SqlCredential $sqlcredential
+			}
+			catch
+			{
+				Write-Warning "Failed to connect to: $instance"
+				continue
+			}
+			
+			$defaults = 'Name', 'Status', 'ContainmentType', 'RecoveryModel', 'CompatibilityLevel', 'Collation', 'Owner', 'EncryptionEnabled'
+			
+			if ($System)
+			{
+				$inputobject = $server.Databases | Where-Object { $_.IsSystemObject -eq $true }
+			}
+			
+			if ($User)
+			{
+				$inputobject = $server.Databases | Where-Object { $_.IsSystemObject -eq $false }
+			}
+			
+			if ($databases)
+			{
+				$inputobject = $server.Databases | Where-Object { $_.Name -in $databases }
+			}
+			
+			if ($state -eq "Online")
+			{
+                $inputobject = $server.Databases | Where-Object { $_.status -eq 'Normal' }
+			}
+			
+			if ($state -eq "Offline")
+			{
```

Comment:
If you do an "IF" more than once for the same property it will be less lines of code to just use the switch statement:

```powershell
switch ($state) {
   "Online" {$inputobject =....}
   "Offline" {$inputobject = ....}
}